### PR TITLE
Create build pipeline

### PR DESCRIPTION
**Pull request by pbourke**

### CODE REVIEW

File: sammo/throttler_test.py [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
+from sammo.throttler import Throttler, AtMost
 
 
 async def simple_job(job_id, throttler, fail=False, delay=0):
     scheduled = time.perf_counter()
     job = await throttler.wait_in_line()
     run = time.perf_counter()
     if delay > 0:
-        await sleep(delay)
+        await asyncio.sleep(delay)
```
Comment (pbourke) [inline sammo/throttler_test.py:16]: @t-schn I think this preserves the functionality of the separate sleep method. The call to `time.sleep` seemed to be adding some non-determinism in the GitHub Actions runner environment, which is much noisier than our dev environments. These test failures only showed up here - everything worked fine for me on my dev machine.

File: sammo/throttler_test.py [hunk 2]
```diff
@@ -35,8 +34,9 @@ async def test_basic_call_limit(n_jobs, completion_time):
         jobs = [g.create_task(simple_job(i, throttler)) for i in range(n_jobs)]
     jobs = [j.result() for j in jobs]
     durations = [j["duration"] for j in jobs]
-    assert min(durations) <= completion_time
-    assert max(durations) <= completion_time
+    # provide a relaxed upper bound for max duration to account for differences
+    # in executors across test environments
+    assert max(durations) <= (completion_time * 2)
```
Comment (pbourke) [inline sammo/throttler_test.py:39]: @t-schn This relaxed upper bound seems to make all environments happy. MacOS seems to be the flakiest of the 3 runners.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        id: setup-python
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Display system and Python versions
+        run: |
+          python -c 'import platform as p; print(f"{p.python_implementation()} {p.python_version()} | {p.system()} {p.release()}")'
```
Comment (daxpryce) [inline .github/workflows/build.yml:37]: is `python --version` not safe enough?

**Comment by pbourke** [comment]
MacOS flakiness continues: https://github.com/microsoft/sammo/actions/runs/7560893210/job/20587901919?pr=11#step:8:121

File: .github/workflows/build.yml (hunk 3 above)
Comment (pbourke) [reply .github/workflows/build.yml:37]: Just wanted to add this as an FYI to track both the python and os versions in case it's relevant for debugging

Comment (daxpryce) [reply .github/workflows/build.yml:37]: fair enough, but this does exist: https://github.com/microsoft/sammo/actions/runs/7560893210/job/20587901480?pr=11#step:1:2 :)

Comment (pbourke) [reply .github/workflows/build.yml:37]: oh that's a good point - I missed that. This step is probably redundant, then

File: sammo/throttler_test.py (hunk 2 above)
Comment (t-schn) [reply sammo/throttler_test.py:39]: Thanks!

**Review by t-schn — APPROVED** [review]
Looks good!

**Pull request merged**